enhancements/network/nodeport-range-configuration.md

```diff
@@ -0,0 +1,125 @@
+---
+title: Make Service NodePort Range Configurable
+authors:
+  - "@abhat"
+reviewers:
+  - "@dcbw"
+  - "@deads2k"
+  - "@smarterclayton"
+  - "@danwinship"
+  - "@knobunc"
+approvers:
+  - "@knobunc"
+creation-date: 2020-07-06
+last-updated: 2020-07-06
+status: implementable
+---
+# Configurability of Service NodePort Range
+This enhancement targets making the service node-port range of an Openshift cluster configurable.
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Service node-port range is a [configurable parameter](https://github.com/openshift/cluster-kube-apiserver-operator/blob/master/bindata/v4.1.0/config/defaultconfig.yaml#L147)
+on the Kubernetes API server. As our customers scale the number of node-port services in their Openshift clusters,
+there is a desire to make this node port range configurable.
+
+## Motivation
+
+Service node-port range is a configurable parameter consumed by the Kubernetes API server as
+a slice of port ranges. It defaults currently to a single port range (30000-32767).
+It is consumed in Openshift via the ConfigMap resource that holds all the api-server configuration.
+In order to allow customers to scale the number of node-port services in the cluster to a value greater
+than that allowed by the default service node-port range, it is important to allow dynamically changing this
+range to allow for more ports. This enhancement focuses on the mechanics of allowing users to configure a
+custom service node-port range for their clusters.
+
+### Goals
+
+- Customers can configure the service node-port range for their cluster during install.
+- Customers can configure the service node-port range for their cluster post install as their needs change.
```

Comment:
The portallocator uses a bitmap, snapshoted in etcd. This implies that you can't have all the API servers active when doing this change, one API server can overwrite the bitmap with the new size. 
Also the master controller run a repair loop that may breal the resizing.